src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -152,6 +153,12 @@ impl Interpreter for CreateTableInterpreter {
 impl CreateTableInterpreter {
     #[async_backtrace::framed]
     async fn create_table_as_select(&self, select_plan: Box<Plan>) -> Result<PipelineBuildResult> {
+        if self.plan.read_only_attach {
```

Comment:
This check aims to?

Reply:
that is a redundant check, let me remove it

Reply:
It's useful to do create table from a attach table in some cases, for example user can convert the attach table to a cloud fuse normal table.
Others LGTM.